### PR TITLE
mappers: changed pdf generator lib

### PR DESCRIPTION
**Pull request by mmadariaga**
Added HTML templates
Added klear help texts
Unified composer files
composer.lock is not ignored anymore
Updated documentation

This PR fixes https://github.com/irontec/ivozprovider/issues/138

### CODE REVIEW

**Comment by Kaian** [comment]
Delta number 055 has already been used in #135 

You should use 056 but #141 also has 056, so please change you're delta number to 057 :smile:

File: library/composer.json [hunk 1]
```diff
@@ -2,12 +2,8 @@
...
-        "psliwa/php-pdf": "~1.2"
-    },
-    "authors": [
-        {
-            "name": "Luis García",
-            "email": "lfgarcia@irontec.com"
-        }
-    ]
+        "knplabs/knp-snappy": "^0.4.3",
+        "h4cc/wkhtmltopdf-amd64": "^0.12.3",
```
Comment (mmadariaga) [inline library/composer.json]: A note for reviewers: i386 binary does not work properly in 64bit debian

File: library/IvozProvider/Gearmand/Invoices/Generator.php [hunk 2]
```diff
@@ -103,15 +91,31 @@ protected function _createInvoice()
...
-        $template = $templateModel->getTemplate();
-        $xml = \IvozProvider\Template\Formatter::format($template, $variables);
+
+        $header = \IvozProvider\Template\Formatter::format($templateModel->getTemplateHeader(), $variables);
+        $body = \IvozProvider\Template\Formatter::format($templateModel->getTemplate(), $variables);
+        $footer = \IvozProvider\Template\Formatter::format($templateModel->getTemplateFooter(), $variables);
 
         $this->_log("Rendering the PDF", \Zend_Log::DEBUG);
-        return $facade->render($xml);
+        $architecture = (2147483647 == PHP_INT_MAX) ? 'i386' : 'amd64';
```
Comment (Kaian) [inline library/IvozProvider/Gearmand/Invoices/Generator.php]: what about [php_uname](http://php.net/manual/es/function.php-uname.php)

`$architecture = (php_uname("m") === "x86_64") ? 'amd64' : 'i386';`

Comment (mmadariaga) [inline library/IvozProvider/Gearmand/Invoices/Generator.php]: You've got it

**Comment by mmadariaga** [comment]
@cruzccl 

> Change your delta to 058 please 

Done

**Pull request merged**